server/src/test/java/org/opensearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -2559,6 +2563,62 @@ public void testApplyContextWithSettingsOverlap() throws IOException {
         }
     }
 
+    public void testApplyCreateIndexRequestWithNoTemplates() throws Exception {
+        BiConsumer<Metadata.Builder, IndexMetadata> mockMetadataTransformer = mock(BiConsumer.class);
+        SystemIndices mockSystemIndices = mock(SystemIndices.class);
+        when(mockSystemIndices.isSystemIndex(anyString())).thenReturn(true);
+
+        ClusterService clusterService = mock(ClusterService.class);
+        ClusterState clusterState = ClusterState.builder(org.opensearch.cluster.ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+            .metadata(Metadata.EMPTY_METADATA)
+            .build();
+
+        ThreadPool threadPool = new TestThreadPool(getTestName());
+
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        when(clusterService.getSettings()).thenReturn(Settings.EMPTY);
+        when(clusterService.getClusterSettings()).thenReturn(clusterSettings);
+        when(clusterService.state()).thenReturn(clusterState);
+
+        MetadataCreateIndexService metadataCreateIndexService = spy(
+            new MetadataCreateIndexService(
+                Settings.EMPTY,
+                clusterService,
+                indicesServices,
+                null,
+                null,
+                createTestShardLimitService(randomIntBetween(1, 1000), false, clusterService),
+                new Environment(Settings.builder().put("path.home", "dummy").build(), null),
+                IndexScopedSettings.DEFAULT_SCOPED_SETTINGS,
+                threadPool,
+                null,
+                mockSystemIndices,
+                true,
+                new AwarenessReplicaBalance(Settings.EMPTY, clusterService.getClusterSettings()),
+                DefaultRemoteStoreSettings.INSTANCE,
+                repositoriesServiceSupplier
+            )
+        );
+
+        metadataCreateIndexService.applyCreateIndexRequest(clusterState, request, false, mockMetadataTransformer);
+        verify(metadataCreateIndexService).applyCreateIndexRequestWithNoTemplates(
+            eq(clusterState),
+            eq(request),
+            eq(false),
+            eq(mockMetadataTransformer)
+        );
+
+        verify(metadataCreateIndexService).applyCreateIndexRequestWithV1Templates(
+            eq(clusterState),
+            eq(request),
+            eq(false),
+            eq(Collections.emptyList()),
+            eq(mockMetadataTransformer)
+        );
```

Comment:
I personally don't think tests like these add a lot of value. You're really just testing that you're calling an internal method that you think you're calling. This isn't verifying that the code is actually doing the thing you expect it to be doing (i.e. that index templates are not actually applied to a system index). Furthermore, this becomes brittle to internal refactorings of the class under test even if the refactoring does not change the behavior of the class at all.

Reply:
> is there any way to create an integration test of the scenario you observed that would validate this fix?

Good call-out. This seems the better approach here.  The repro in the original issue should be easy to set up in an integ test, the only hitch here is making sure `systemIndices` is populated with whatever test index we're choosing, unless the test is isolated enough that we can use one of the actual indices specified.

Reply:
Having in YAML spec or IT test case would definitely be helpful

Reply:
@pyek-bot Here is an example test that asserts that deprecation warnings are emitted when a user writes to a system index: https://github.com/opensearch-project/OpenSearch/blob/4e45e44742bc30ba7571ea4ba88de2c1cd1c6d88/qa/smoke-test-http/src/test/java/org/opensearch/http/SystemIndexRestIT.java#L82

Reply:
Thanks @pyek-bot, the IT looks good!

I'd actually be in favor of removing this unit test. I don't think it adds a lot of value, and it does have some cost by exposing internal details of the class under test that might change in future refactorings.

Reply:
> I'd actually be in favor of removing this unit test. I don't think it adds a lot of value, and it does have some cost by exposing internal details of the class under test that might change in future refactorings.

+1, and put it back to a private method.

@cwperks implied earlier he might try to refactor all that common code anyway.